backend/src/main/repository/PromotionRepository.ts

```diff
@@ -71,6 +72,24 @@ export class PromotionRepository extends Repository<Promotion> {
       );
     }
 
+    if (promotionQuery?.dayOfWeek) {
+      // use a subQuery so that we still return all the schedules of a promotion
+      queryBuilder.andWhere(
+        (qb) => {
+          const subQuery = qb
+            .subQuery()
+            .select('S.promotionId')
+            .from(Schedule, 'S')
+            .where('S.dayOfWeek = :dayOfWeek')
+            .getQuery();
+          return 'promotion.id in ' + subQuery;
+        },
+        {
+          dayOfWeek: promotionQuery.dayOfWeek,
+        }
+      );
+    }
```

Comment:
This translates to SQL query like. The reason we want to use a subquery is that we still want to return all the schedules associated with a promotion
```
select * from promotion 
inner join discount on promotion.id = discount."promotionId"
inner join schedule on promotion.id = schedule."promotionId"
where promotion.id in (select S."promotionId" from schedule S where S."dayOfWeek" = :dayOfWeek);
```